IOHexperimenter/Cpp.md

```diff
@@ -72,52 +76,81 @@ const auto &suite_factory = ioh::suite::SuiteRegistry<ioh::problem::Real>::insta
 const auto bbob = suite_factory.create("BBOB", {1, 2}, {1, 3}, {5, 10});
 ```
 
-In the given example, a random search is applied across all problems within a loop :
+In the example, a random search is applied across all the problems within a loop :
 ```cpp
 for (const auto &problem : *bbob) :
 {
   ...
 }
 ```
-The variable `problem` is a pointer of __Problem__ class upper introduced.
+Usage of the `problem` class is introduced [above](#using-individual-problems).
 
 
 <a name = "using-logger"></a>
 ## Using Logger
 
-A default logger is provided to record evaluation information during the optimization process into csv files. The following statement declares a logger, which outputs files into a folder named as `logger_example` at the directory `./`. The algorithm name `random_search` and the algorithm information `a random search for ...` will be attached into the csv files.
+The default logger records evaluation information during the optimization process into csv files. An example of using the default logger is provided in [logger_example.h](https://github.com/IOHprofiler/IOHexperimenter/blob/master/example/logger_example.h)
+
+
+We declare a default logger using the following statement.
 ```cpp
-auto l = ioh::logger::Default("./", "logger_example", "random_search", "a random search for testing the bbob suite");
+auto logger = ioh::logger::Default(
+        fs::current_path(), /* output_directory : fs::current_path() -> working directory */
+        "folder_name", /* folder_name */
+        "random_search", /* algorithm_name : "random search'*/
+        "a random search" /* algorithm_info : "a random search for testing the bbob suite" */
+        );
 ```
 
-By default, evaluations when the best-found fitness is updated will be recorded into csv files. This can be changed by passing arguments to the constructor. In practice,
+We attach the `logger` to the `suite`, to perform logging process during optimization.
+```cpp
+suite->attach_logger(logger);
+```
+
+Alternatively, we can attach the logger to each new problem in a suite manually using
+```cpp
+problem->attach_logger(logger);
```

Comment:
It might be better to use the onemax or sphere problem you defined earlier here, so this problem variable is defined